src/constructs/lambda/lambda.test.ts

```diff
@@ -27,12 +28,30 @@ describe("GuLambdaFunction", () => {
       runtime: Runtime.NODEJS_12_X,
       rules: [
         {
-          frequency: Duration.days(7),
+          schedule: Schedule.rate(Duration.days(7)),
           description: "run every week",
         },
       ],
     });
 
     expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
   });
+
+  it("should create a lambda function with an api gateway", () => {
+    const stack = new Stack();
+
+    new GuLambdaFunction(stack, "lambda", {
```

Comment:
We probably want `GuLambdaFunction` to take in a `GuStack` as it's first argument, rather than a `Construct` as this will provide the stack and stage params and tags for free.